bin/dsh

```diff
@@ -736,7 +753,48 @@ vmstat ()
 	vboxmanage metrics query "$id" $metrics
 }
 
-#-------------------------- Installation and updates -----------------------------
+# Add key to ssh-agent or run ssh-add with provided @param
+# @param $1 -D, -l or path to custom key
+ssh_add () {
+	# Check if ssh-agent container is running
+	local running=$(docker inspect --format="{{ .State.Running }}" ssh-agent 2> /dev/null)
+	if [ $? -eq 0 ] && [ "$running" == "true" ]; then
+		case "$1" in
+			"")
+				if docker exec -it ssh-agent ssh-add -l > /dev/null; then
+					return; # Some key is present
+				fi
+				if is_linux; then
+					local fullpath="${HOME}/.ssh/id_rsa"
+				else
+					local fullpath="/.home/.ssh/id_rsa"
+				fi
+				if [ ! -f "$fullpath" ]; then
+					echo "$fullpath: No such file or directory"
+					echo "Trying to use id_dsa"
+					local fullpath=${fullpath/id_rsa/id_dsa}
+				fi
+				;;
+			-D) 
+				docker run --rm --volumes-from=ssh-agent -it blinkreaction/ssh-agent:${DRUDE_ITAG} ssh-add -D
+				exit 1
+				;;
+			-l) 
+				docker run --rm --volumes-from=ssh-agent -it blinkreaction/ssh-agent:${DRUDE_ITAG} ssh-add -l
+				exit 1
+				;;
+			*) 
+				local fullpath=$1
+				;;
+		esac
+		local directory=$(dirname "$fullpath")
+		local filename=$(basename "$fullpath")
```

Comment:
I dont see a reason to break this into pieces to only join back again 2 lines below